pkg/controller/watermarkpodautoscaler/watermarkpodautoscaler_controller.go

```diff
@@ -65,9 +85,12 @@ func newReconciler(mgr manager.Manager) (reconcile.Reconciler, error) {
 		nil,
 		external_metrics.NewForConfigOrDie(clientConfig),
 	)
+
+	podLister := initializePodInformer(clientConfig)
```

Comment:
to be about to pass a stop channel to the podLister. I see two options:
* You can add add a `context.Context`(or just a `stop chan`) in the `ReconcileWatermarkPodAutoscaler` struct. the `ReconcileWatermarkPodAutoscaler.Start(...)` receive a stop channel. so you can cascade after the stop event to the internal channel.
* you can initialize the pod informer only when you start the `ReconcileWatermarkPodAutoscaler`.

Reply:
I moved the channel outside of the initializePodInfomer method.
I will implement more logic to handle the cancelation in the reconcile loop in a subsequent PR.